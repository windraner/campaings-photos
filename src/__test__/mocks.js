export const markersList = [
  {
    id: 'all',
    name: 'Show all',
    isChecked: false,
    type: 'all',
    count: 0
  },
  {
    id: 'pending',
    name: 'Pending',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'approved',
    name: 'Approved',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'rejected',
    name: 'Rejected',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'rightsRequested',
    name: 'Rights requested',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'sticly',
    name: 'Sticly',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'topPhotos',
    name: 'Top photos',
    isChecked: false,
    type: 'secondary',
    count: 0
  },
  {
    id: 'unhealthy',
    name: 'Unhealthy products',
    isChecked: false,
    type: 'secondary',
    count: 0
  },
];
