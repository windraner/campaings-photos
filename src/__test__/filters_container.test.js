import React from 'react';
import FiltersContainer from '../components/filters_container/filters_container';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { markersList } from './mocks';

configure({ adapter: new Adapter() });

const mock = {
  searchInput: '',
  markersList: markersList,
  isFilterShown: true,
  stateHandler() {},
};

describe('FiltersContainer', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<FiltersContainer {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
