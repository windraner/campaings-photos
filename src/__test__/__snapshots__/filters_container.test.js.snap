// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersContainer matches the snapshot 1`] = `
<div
  className="campaigns-photos__filter-container"
>
  <CustomInput
    searchInput=""
    stateHandler={[Function]}
  />
  <div
    className="campaigns-photos_mt-32"
  >
    <FilterCheckLine
      item={
        Object {
          "count": 0,
          "id": "all",
          "isChecked": false,
          "name": "Show all",
          "type": "all",
        }
      }
      key="all"
      markersList={
        Array [
          Object {
            "count": 0,
            "id": "all",
            "isChecked": false,
            "name": "Show all",
            "type": "all",
          },
          Object {
            "count": 0,
            "id": "pending",
            "isChecked": false,
            "name": "Pending",
            "type": "main",
          },
          Object {
            "count": 0,
            "id": "approved",
            "isChecked": false,
            "name": "Approved",
            "type": "main",
          },
          Object {
            "count": 0,
            "id": "rejected",
            "isChecked": false,
            "name": "Rejected",
            "type": "main",
          },
          Object {
            "count": 0,
            "id": "rightsRequested",
            "isChecked": false,
            "name": "Rights requested",
            "type": "main",
          },
          Object {
            "count": 0,
            "id": "sticly",
            "isChecked": false,
            "name": "Sticly",
            "type": "main",
          },
          Object {
            "count": 0,
            "id": "topPhotos",
            "isChecked": false,
            "name": "Top photos",
            "type": "secondary",
          },
          Object {
            "count": 0,
            "id": "unhealthy",
            "isChecked": false,
            "name": "Unhealthy products",
            "type": "secondary",
          },
        ]
      }
      stateHandler={[Function]}
    />
    <div
      className="campaigns-photos_mt-32"
    >
      <FilterCheckLine
        item={
          Object {
            "count": 0,
            "id": "pending",
            "isChecked": false,
            "name": "Pending",
            "type": "main",
          }
        }
        key="pending"
        markersList={
          Array [
            Object {
              "count": 0,
              "id": "all",
              "isChecked": false,
              "name": "Show all",
              "type": "all",
            },
            Object {
              "count": 0,
              "id": "pending",
              "isChecked": false,
              "name": "Pending",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "approved",
              "isChecked": false,
              "name": "Approved",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rejected",
              "isChecked": false,
              "name": "Rejected",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rightsRequested",
              "isChecked": false,
              "name": "Rights requested",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "sticly",
              "isChecked": false,
              "name": "Sticly",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "topPhotos",
              "isChecked": false,
              "name": "Top photos",
              "type": "secondary",
            },
            Object {
              "count": 0,
              "id": "unhealthy",
              "isChecked": false,
              "name": "Unhealthy products",
              "type": "secondary",
            },
          ]
        }
        stateHandler={[Function]}
      />
      <FilterCheckLine
        item={
          Object {
            "count": 0,
            "id": "approved",
            "isChecked": false,
            "name": "Approved",
            "type": "main",
          }
        }
        key="approved"
        markersList={
          Array [
            Object {
              "count": 0,
              "id": "all",
              "isChecked": false,
              "name": "Show all",
              "type": "all",
            },
            Object {
              "count": 0,
              "id": "pending",
              "isChecked": false,
              "name": "Pending",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "approved",
              "isChecked": false,
              "name": "Approved",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rejected",
              "isChecked": false,
              "name": "Rejected",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rightsRequested",
              "isChecked": false,
              "name": "Rights requested",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "sticly",
              "isChecked": false,
              "name": "Sticly",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "topPhotos",
              "isChecked": false,
              "name": "Top photos",
              "type": "secondary",
            },
            Object {
              "count": 0,
              "id": "unhealthy",
              "isChecked": false,
              "name": "Unhealthy products",
              "type": "secondary",
            },
          ]
        }
        stateHandler={[Function]}
      />
      <FilterCheckLine
        item={
          Object {
            "count": 0,
            "id": "rejected",
            "isChecked": false,
            "name": "Rejected",
            "type": "main",
          }
        }
        key="rejected"
        markersList={
          Array [
            Object {
              "count": 0,
              "id": "all",
              "isChecked": false,
              "name": "Show all",
              "type": "all",
            },
            Object {
              "count": 0,
              "id": "pending",
              "isChecked": false,
              "name": "Pending",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "approved",
              "isChecked": false,
              "name": "Approved",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rejected",
              "isChecked": false,
              "name": "Rejected",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rightsRequested",
              "isChecked": false,
              "name": "Rights requested",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "sticly",
              "isChecked": false,
              "name": "Sticly",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "topPhotos",
              "isChecked": false,
              "name": "Top photos",
              "type": "secondary",
            },
            Object {
              "count": 0,
              "id": "unhealthy",
              "isChecked": false,
              "name": "Unhealthy products",
              "type": "secondary",
            },
          ]
        }
        stateHandler={[Function]}
      />
      <FilterCheckLine
        item={
          Object {
            "count": 0,
            "id": "rightsRequested",
            "isChecked": false,
            "name": "Rights requested",
            "type": "main",
          }
        }
        key="rightsRequested"
        markersList={
          Array [
            Object {
              "count": 0,
              "id": "all",
              "isChecked": false,
              "name": "Show all",
              "type": "all",
            },
            Object {
              "count": 0,
              "id": "pending",
              "isChecked": false,
              "name": "Pending",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "approved",
              "isChecked": false,
              "name": "Approved",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rejected",
              "isChecked": false,
              "name": "Rejected",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rightsRequested",
              "isChecked": false,
              "name": "Rights requested",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "sticly",
              "isChecked": false,
              "name": "Sticly",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "topPhotos",
              "isChecked": false,
              "name": "Top photos",
              "type": "secondary",
            },
            Object {
              "count": 0,
              "id": "unhealthy",
              "isChecked": false,
              "name": "Unhealthy products",
              "type": "secondary",
            },
          ]
        }
        stateHandler={[Function]}
      />
      <FilterCheckLine
        item={
          Object {
            "count": 0,
            "id": "sticly",
            "isChecked": false,
            "name": "Sticly",
            "type": "main",
          }
        }
        key="sticly"
        markersList={
          Array [
            Object {
              "count": 0,
              "id": "all",
              "isChecked": false,
              "name": "Show all",
              "type": "all",
            },
            Object {
              "count": 0,
              "id": "pending",
              "isChecked": false,
              "name": "Pending",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "approved",
              "isChecked": false,
              "name": "Approved",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rejected",
              "isChecked": false,
              "name": "Rejected",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rightsRequested",
              "isChecked": false,
              "name": "Rights requested",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "sticly",
              "isChecked": false,
              "name": "Sticly",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "topPhotos",
              "isChecked": false,
              "name": "Top photos",
              "type": "secondary",
            },
            Object {
              "count": 0,
              "id": "unhealthy",
              "isChecked": false,
              "name": "Unhealthy products",
              "type": "secondary",
            },
          ]
        }
        stateHandler={[Function]}
      />
    </div>
    <div
      className="campaigns-photos_mt-32"
    >
      <FilterCheckLine
        item={
          Object {
            "count": 0,
            "id": "topPhotos",
            "isChecked": false,
            "name": "Top photos",
            "type": "secondary",
          }
        }
        key="topPhotos"
        markersList={
          Array [
            Object {
              "count": 0,
              "id": "all",
              "isChecked": false,
              "name": "Show all",
              "type": "all",
            },
            Object {
              "count": 0,
              "id": "pending",
              "isChecked": false,
              "name": "Pending",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "approved",
              "isChecked": false,
              "name": "Approved",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rejected",
              "isChecked": false,
              "name": "Rejected",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rightsRequested",
              "isChecked": false,
              "name": "Rights requested",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "sticly",
              "isChecked": false,
              "name": "Sticly",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "topPhotos",
              "isChecked": false,
              "name": "Top photos",
              "type": "secondary",
            },
            Object {
              "count": 0,
              "id": "unhealthy",
              "isChecked": false,
              "name": "Unhealthy products",
              "type": "secondary",
            },
          ]
        }
        stateHandler={[Function]}
      />
      <FilterCheckLine
        item={
          Object {
            "count": 0,
            "id": "unhealthy",
            "isChecked": false,
            "name": "Unhealthy products",
            "type": "secondary",
          }
        }
        key="unhealthy"
        markersList={
          Array [
            Object {
              "count": 0,
              "id": "all",
              "isChecked": false,
              "name": "Show all",
              "type": "all",
            },
            Object {
              "count": 0,
              "id": "pending",
              "isChecked": false,
              "name": "Pending",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "approved",
              "isChecked": false,
              "name": "Approved",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rejected",
              "isChecked": false,
              "name": "Rejected",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "rightsRequested",
              "isChecked": false,
              "name": "Rights requested",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "sticly",
              "isChecked": false,
              "name": "Sticly",
              "type": "main",
            },
            Object {
              "count": 0,
              "id": "topPhotos",
              "isChecked": false,
              "name": "Top photos",
              "type": "secondary",
            },
            Object {
              "count": 0,
              "id": "unhealthy",
              "isChecked": false,
              "name": "Unhealthy products",
              "type": "secondary",
            },
          ]
        }
        stateHandler={[Function]}
      />
    </div>
  </div>
</div>
`;
